async_simple/coro/Task.h

```diff
@@ -139,8 +139,8 @@ struct TaskPromiseBase {
     }
     void unhandled_exception() const noexcept {}
 
-    void* operator new(size_t sz) noexcept { return ::operator new(sz); }
-    void operator delete(void* p, size_t sz) noexcept {
+    void* operator new(size_t sz) { return ::operator new(sz); }
+    void operator delete(void* p, size_t sz) {
```

Comment:
Why do we need to remove the `noexcept` specifier?

Reply:
Not compile in gcc, Task::promise_type do not have a get_return_object_on_allocation_failure().

Reply:
It would be better to add `get_return_object_on_allocation_failure()` methods. It could return a Lazy with an exception shows it fails on allocation failure.

Reply:
Task do not have an exception_ptr member, which is different from Lazy.

Reply:
How about adding `exception_ptr` for Task? Would it be too hard?

Reply:
`Task` was created for in-memory applications, maybe we could remove it since nobody is using it.

Reply:
If nobody uses it, I agree we should remove it.

Reply:
Will fix in #29

Reply:
OK